### ISSUE TITLE
back to showing errors for already-copied chunks during restart

### ISSUE THREAD

**Issue by jti-lanl** [pftool/pftool#27]
This was fixed.  Now it's broken again.  Looks like the state of the file-handle/object-stream/IO-buf is not matching the old set of conditions that we were looking for.  Maybe it is getting screwed up, as well?

**Comment by jti-lanl** [comment]
[mysteriously working again?]

What this is about is the output that was fixed in 72303b5:  when recopying old chunks that were previously copied but didn't yet get accounted for in CTM, we had fixed things to get output like this:

> RANK   3: INFO  DATACOPY *Copied /src/f01 offs 0 len 1073738881 to /marfs/ns/f01
> RANK   3: INFO  DATACOPY *Copied /src/f01 offs 1073738881 len 1073738881 to /marfs/ns/f01
> RANK   3: INFO  DATACOPY Copied /src/f01 offs 2147477762 len 1073738881 to /marfs/ns/f01

Where the asterisk indicates "deemed success", in which the write failed, but a subsequent HEAD determined that it failed because the chunk already exists.

This seemed to have stopped working.  We'd get the old error message: "wrote -1 bytes instead of [blah]".  The problem was that flags that should have been getting set in the file-handle in the MARFS_Path object, were no longer getting set, which prevented us detecting the situation for the HEAD request.

These flags are set in the file-handle in marfs_open_at_offset(), called from MARFS_Path::open().  In gdb we could see (a pointer to) the file-handle member of MARFS_Path, passed into marfs_open_at_offset(), and the flags changed there, but on return they no longer seemed to have been set.  However, other functions that were also given a pointer to the FH could also see that the flags were set.

Tried checkout/build/test through a sequence of old commits, looking for the problem, and got all the way to the present without finding it.  Everything seems to work correctly again.  Perhaps there were old files affecting the build, which got purged during that exercise?

Keeping notes here, in case we run into it again.
